docs/aws/audit/redshiftmonitoring/rules/emr_instance_type_generation_remediation.mdx

```diff
@@ -1,6 +1,261 @@
+2. Navigate to the Redshift dashboard by selecting "Services" from the top menu, then selecting "Redshift" under the "Database" category.
+3. In the Redshift dashboard, select "Clusters" from the left-hand menu. This will display a list of all your Redshift clusters.
+4. For each cluster, check the "Node type" column. If the node type is not the latest generation (for example, dc2.large), then the EMR instances are not using the latest generation in Redshift.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the Redshift clusters.
+
+2. Once the AWS CLI is set up, you can list all the Redshift clusters using the following command:
+
+   ```
+   aws redshift describe-clusters --query 'Clusters[*].[ClusterIdentifier,NodeType]' --output text
+   ```
+   This command will return a list of all Redshift clusters along with their node types.
+
+3. To check if the Redshift clusters are using the latest generation of EMR instances, you need to compare the NodeType of each cluster with the latest generation of EMR instances. As of now, the latest generation of EMR instances is dc2.
+
+4. You can use the following Python script to automate the process:
+
+   ```python
+   import subprocess
+   import json
+
+   def get_clusters():
+       command = "aws redshift describe-clusters --query 'Clusters[*].[ClusterIdentifier,NodeType]'"
+       process = subprocess.Popen(command, stdout=subprocess.PIPE, shell=True)
+       output, error = process.communicate()
+       return json.loads(output)
+
+   def check_latest_generation(clusters):
+       latest_generation = 'dc2'
+       for cluster in clusters:
+           if not cluster[1].startswith(latest_generation):
+               print(f"Cluster {cluster[0]} is not using the latest generation of EMR instances")
+
+   if __name__ == "__main__":
+       clusters = get_clusters()
+       check_latest_generation(clusters)
+   ```
+   This script first gets a list of all Redshift clusters and their node types. Then it checks if the node type of each cluster starts with 'dc2', which is the latest generation of EMR instances. If a cluster is not using the latest generation, it prints a message with the cluster identifier.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary AWS SDK for Python (Boto3) modules in your Python script. You will need the 'boto3' and 'botocore' modules to interact with AWS services.
+
+```python
+import boto3
+from botocore.exceptions import BotoCoreError, ClientError
+```
+
+2. Create a session using your AWS credentials. Replace 'aws_access_key_id', 'aws_secret_access_key', and 'aws_session_token' with your actual AWS credentials.
+
+```python
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>